framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -537,14 +539,26 @@ public Boolean visitTypeArgs(
                 return true;
             }
 
-            if (supertypeTypeArgs.size() > 0) {
-                for (int i = 0; i < supertypeTypeArgs.size(); i++) {
-                    final AnnotatedTypeMirror superTypeArg = supertypeTypeArgs.get(i);
-                    final AnnotatedTypeMirror subTypeArg = subtypeTypeArgs.get(i);
-                    if (!compareTypeArgs(
-                            subTypeArg, superTypeArg, supertypeRaw, subtypeRaw, visited)) {
-                        return false;
-                    }
+            final TypeElement supertypeElem =
+                    (TypeElement) supertype.getUnderlyingType().asElement();
+            List<Integer> covariantArgIndexes = null;
+            AnnotationMirror covam =
+                    supertype.atypeFactory.getDeclAnnotation(supertypeElem, Covariant.class);
+            if (covam != null) {
+                covariantArgIndexes =
+                        AnnotationUtils.getElementValueArray(covam, "value", Integer.class, false);
+            }
+
+            for (int i = 0; i < supertypeTypeArgs.size(); i++) {
+                final AnnotatedTypeMirror superTypeArg = supertypeTypeArgs.get(i);
+                final AnnotatedTypeMirror subTypeArg = subtypeTypeArgs.get(i);
+                final boolean covariant =
+                        this.covariantTypeArgs
```

Comment:
I didn't find a use of the constructor that sets this field to true, not within the Checker Framework and not in sparta.
When does it make sense to set `covariantTypeArgs` true? It seems rather risky to allow co-variance for all types.
Does anyone know a reason? Otherwise, I would like to remove it.